src/main/kotlin/org/jitsi/nlj/transform/node/outgoing/RetransmissionSender.kt

```diff
@@ -48,39 +49,48 @@ class RetransmissionSender : TransformerNode("Retransmission sender") {
     private val rtxStreamSeqNums: MutableMap<Long, Int> = mutableMapOf()
 
     private var numRetransmissionsRequested = 0
-    private var numRetransmittedPackets = 0
+    private var numRetransmittedRtxPackets = 0
+    private var numRetransmittedPlainPackets = 0
 
     override fun transform(packetInfo: PacketInfo): PacketInfo? {
         val rtpPacket = packetInfo.packetAs<RtpPacket>()
-        logger.cdebug { "${hashCode()} retransmitting packet with original ssrc " +
-                "${rtpPacket.ssrc}, original sequence number ${rtpPacket.sequenceNumber} and original " +
-                "payload type: ${rtpPacket.payloadType}" }
         numRetransmissionsRequested++
-        val rtxSsrc = associatedSsrcs[rtpPacket.ssrc] ?: run {
-            logger.cerror { "${hashCode()} could not find an associated RTX ssrc for original packet ssrc " +
-                    rtpPacket.ssrc }
-            packetDiscarded(packetInfo)
-            return null
-        }
-        val rtxPt = associatedPayloadTypes[rtpPacket.payloadType.toPositiveInt()] ?: run {
-            logger.cerror { "${hashCode()} could not find an associated RTX payload type for original payload type " +
-                    rtpPacket.payloadType }
-            packetDiscarded(packetInfo)
-            return null
+        var rtxRetransmission = false
+        val rtxSsrc = associatedSsrcs[rtpPacket.ssrc]
+        if (rtxSsrc != null) {
+            val rtxPt = associatedPayloadTypes[rtpPacket.payloadType.toPositiveInt()]
+            if (rtxPt != null) {
+                // Get a default value of 1 to start if it isn't present in the map.  If it is present
+                // in the map, get the value and increment it by 1
+                val rtxSeqNum = rtxStreamSeqNums.merge(rtxSsrc, 1, Integer::sum)!!
+
+                RtxPacket.addOriginalSequenceNumber(rtpPacket)
+
+                rtpPacket.ssrc = rtxSsrc
+                rtpPacket.payloadType = rtxPt
+                rtpPacket.sequenceNumber = rtxSeqNum
+                logger.cdebug { "${hashCode()} sending RTX packet with ssrc $rtxSsrc with pt $rtxPt and seqNum " +
+                        "$rtxSeqNum with original ssrc ${rtpPacket.ssrc}, original sequence number " +
+                        "${rtpPacket.sequenceNumber} and original payload type: ${rtpPacket.payloadType}" }
+                packetInfo.resetPayloadVerification()
+                numRetransmittedRtxPackets++
+                rtxRetransmission = true
+            } else {
+                logger.cwarn {
+                    "${hashCode()} could not find an associated RTX payload type for original payload type " +
```

Comment:
Do we want this as a warning? It happens whenever the remote side doesn't support RTX, right?

Reply:
we hit this sometimes

Reply:
Do we not hit it on every retransmitted packet when the receiver is Firefox?

Reply:
you're right and that was the case prior to 01fb98f. in 01fb98f i switched the guards (we now check for the rtx pt first) so this warning should no longer be hit, in theory

Reply:
Oh, I missed that. Wouldn't we hit it with a Firefox sender and Chrome receiver now?

Reply:
you're right, i scraped it